### PR TITLE
Changed category Button To Select

### LINKED ISSUE #204: bug: categories bar goes out of screen and cannot add more categories
**Describe the bug**
the categories bar goes out of screen and not responsive, also if more tracks are added then we cannot add more tracks to it size there is not enough space

**To Reproduce**
1. go to dailycode
2. make the tab size smaller or view from a smaller screen

**Expected behavior**
should typically be a select tab instead of buttons

**Screenshots or GIFs**

https://github.com/code100x/daily-code/assets/136328410/f5628268-f45a-4df2-adab-3e3fd17177f4

**Info:**
Version 123.0.6312.123 (Official Build) (arm64)

### PR DESCRIPTION
**Pull request by nischal-shetty2**
fixes: #204
added select tag and can now add more categories in the future when required

https://github.com/code100x/daily-code/assets/136328410/d6141f2a-ef5b-4d3d-b442-25d205d1d8b5

### CODE REVIEW

**Comment by nischal-shetty2** [comment]
@hkirat this is the one that was originally submitted which you reviewed

File: packages/ui/src/Categories.tsx [hunk 1]
```diff
@@ -1,29 +1,37 @@
...
-          key={category.category}
-          variant="ghost"
-          onClick={() => handleFilterButton(category.category)}
-          className={selectedCategory == category.category ? "bg-gray-100 dark:bg-slate-700" : ""}
-        >
-          {category.category}
-        </Button>
-      ))}
+    <div className="flex justify-evenly mx-auto">
+      <Select
```
Comment (hkirat) [inline packages/ui/src/Categories.tsx:19]: Can we do this conditionally?
If screen width is less than lg, then maybe render the select?

**Review by hkirat — CHANGES_REQUESTED** [review]

File: packages/ui/src/Categories.tsx (hunk 1 above)
Comment (nischal-shetty2) [reply packages/ui/src/Categories.tsx:19]: yes, on it

Comment (nischal-shetty2) [reply packages/ui/src/Categories.tsx:19]: @hkirat making it less than lg will still have overflow, ill do it at xl?

Comment (hkirat) [reply packages/ui/src/Categories.tsx:51]: yup

**Pull request merged**